servers/src/main/java/tachyon/worker/block/meta/StorageDir.java

```diff
@@ -60,6 +62,48 @@ public StorageDir(StorageTier tier, int dirIndex, long capacityBytes, String dir
     mBlockIdToBlockMap = new HashMap<Long, BlockMeta>(200);
     mBlockIdToTempBlockMap = new HashMap<Long, TempBlockMeta>(200);
     mUserIdToTempBlockIdsMap = new HashMap<Long, Set<Long>>(200);
+
+    initializeMeta();
+  }
+
+  /**
+   * Only the File that is file and filename can be parsed into Long will be accept
+   */
+  class MetaFileFilter implements FileFilter {
+    public boolean accept(File path) {
+      if (!path.isFile()) {
+        return false;
+      }
+      try {
+        Long.valueOf(path.getName());
+        return true;
+      } catch (NumberFormatException nfe) {
+        return false;
+      }
+    }
+  }
+
+  /**
+   * Initialize meta data for existing blocks in this StorageDir
+   *
+   * Only paths satisfying the contract defined in {@link BlockMetaBase#commitPath()} are legal,
+   * should be in format like {dir}/{blockId}. others are ignored.
+   */
+  private void initializeMeta() {
+    File dir = new File(mDirPath);
+    FileFilter filter = new MetaFileFilter();
+    File[] files = dir.listFiles(filter);
```

Comment:
Currently, we are listing all "good" filenames and continue working on these files. On the other hand, I prefer to also LOG error those "bad" filenames---as they are not supposed to show there. Proceeding silently with those "bad" files inside directory is risky. 

So at least LOG error them.